dojo/tools/bugcrowd_api/api_client.py

```diff
@@ -0,0 +1,140 @@
+import requests
+from urllib.parse import urlencode
+from dojo.models import Tool_Type
+
+
+class BugcrowdAPI:
+    """
+    A simple client for the bugcrowd.io API
+    """
+
+    bugcrowd_api_url = "https://api.bugcrowd.com"
+    default_headers = {
+        "Accept": "application/vnd.bugcrowd+json",
+        "User-Agent": "DefectDojo",
+        "Bugcrowd-Version": "2021-10-28",
+    }
+
+    def __init__(self, tool_config):
+        tool_type, _ = Tool_Type.objects.get_or_create(name="Bugcrowd API")
+
+        self.session = requests.Session()
+        if tool_config.authentication_type == "API":
+            self.api_token = tool_config.api_key
+            self.session.headers.update(
+                {"Authorization": "Token {}".format(self.api_token)}
+            )
+            self.session.headers.update(self.default_headers)
+        else:
+            raise Exception(
+                "bugcrowd Authentication type {} not supported".format(
+                    tool_config.authentication_type
+                )
+            )
+
+    def get_findings(self, program, target):
+        """
+        Returns the findings in a paginated iterator for a given bugcrowd program and target, if target is *, everything is returned
+        :param program:
+        :param target:
+        :return:
+        """
+        params_default = {
+            "filter[duplicate]": "false",
+            "filter[program]": program,
+            "page[limit]": 100,
+            "page[offset]": 0,
+            "include": "monetary_rewards,target",
+            "sort": "submitted-desc",
+        }
+
+        if target:
+            params = params_default
+            params["filter[target]"] = target
+            params_encoded = urlencode(params)
+        else:
+            params_encoded = urlencode(params_default)
+
+        next = "{}/submissions?{}".format(self.bugcrowd_api_url, params_encoded)
+        while next != "":
+            response = self.session.get(url=next)
+
+            if response.ok:
+                data = response.json()
+                if len(data["data"]) != 0:
+                    yield data["data"]
+
+                # When we hit the end of the submissions, break out
+                if len(data["data"]) == 0:
+                    next = ""
+                    break
+
+                # Otherwise, keep updating next link
+                next = "{}{}".format(self.bugcrowd_api_url, data["links"]["next"])
+            else:
+                next = "over"
+                raise Exception(
+                    "Unable to get bugcrowd submissions due to {} - {}".format(
+                        response.status_code, response.content.decode("utf-8")
+                    )
+                )
```

Comment:
please, as shared in IRL review, use `response.raise_for_status()` instead of shadowing the root exception.

Reply:
Done